src/passwordify/main.go

```diff
@@ -0,0 +1,184 @@
+// Copyright (2012) Sandia Corporation.
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"golang.org/x/crypto/ssh/terminal"
+	"io"
+	"io/ioutil"
+	log "minilog"
+	"os"
+	"os/exec"
+	"syscall"
+)
+
+var (
+	f_keys = flag.String("keys", "", "authorized_keys formatted file to install for root")
+)
+
+func usage() {
+	fmt.Println("usage: passwordify [option]... [source initramfs] [destination initramfs]")
+	flag.PrintDefaults()
+}
+
+func main() {
+	flag.Usage = usage
+	flag.Parse()
+
+	log.Init()
+
+	if flag.NArg() != 2 {
+		usage()
+		os.Exit(1)
+	}
+
+	source := flag.Arg(0)
+	destination := flag.Arg(1)
+
+	// Make sure the source exists
+	if _, err := os.Stat(source); os.IsNotExist(err) {
+		log.Fatalln("cannot find source initramfs", source, ":", err)
+	}
+
+	// Working directory
+	tdir, err := ioutil.TempDir("", "passwordify")
+	if err != nil {
+		log.Fatalln("Cannot create tempdir:", err)
+	}
+
+	// Unpack initrd
+	initrdCommand := fmt.Sprintf("cd %v && zcat %v | cpio -idmv", tdir, source)
+	err = runscript(initrdCommand)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	// Set password
+	p := process("chroot")
+	cmd := &exec.Cmd{
+		Path: p,
+		Args: []string{
+			p,
+			tdir,
+			"passwd",
+		},
+		Env: nil,
+		Dir: "",
+	}
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	stdin, err := cmd.StdinPipe()
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	go func() {
+		defer stdin.Close()
+		fmt.Printf("Enter new root password: ")
```

Comment:
Could put this in a for loop so we keep reading until we get two passwords that match (or an error).